policy-controller/k8s/index/src/defaults.rs

```diff
@@ -80,32 +85,35 @@ impl DefaultPolicy {
         config: &ClusterInfo,
     ) -> HashMap<AuthorizationRef, ClientAuthorization> {
         let mut authzs = HashMap::default();
-        if let DefaultPolicy::Allow {
-            authenticated_only,
-            cluster_only,
-        } = self
-        {
-            let authentication = if authenticated_only {
-                ClientAuthentication::TlsAuthenticated(vec![IdentityMatch::Suffix(vec![])])
-            } else {
-                ClientAuthentication::Unauthenticated
-            };
-            let networks = if cluster_only {
-                config.networks.iter().copied().map(Into::into).collect()
-            } else {
-                vec![
-                    "0.0.0.0/0".parse::<IpNet>().unwrap().into(),
-                    "::/0".parse::<IpNet>().unwrap().into(),
-                ]
-            };
-            authzs.insert(
-                AuthorizationRef::Default(self.as_str()),
-                ClientAuthorization {
-                    authentication,
-                    networks,
-                },
-            );
+        let (authenticated_only, cluster_only) = match self {
```

Comment:
I find it harder to follow this kind of thing when there are early returns.  What about something like:
```
match self {
  DefaultPolicy::Allow {
    authenticated_only,
    cluster_only,
  } => {
    let authentication = if authenticated_only { ... } else { ... };
    let networks = if cluster_only { ... } else { ... };
    authzs.insert( ... );
  },
  DefaultPolicy::Deny => {},
  DefaultPolicy::Audit => {
    let authentication = ...
    let networks = ...
    authzs.insert( ... );
  },
}
```

TIOLI

Reply:
Yeah that return was a little awkward. We can't use `DefaultPolicy::Deny => {},` because that unit type clashes with the type from the other arms. I've pushed a refactoring using `if` branches instead.

Reply:
I think all the branches have unit type (`authz.insert(...);`) but yeah, if branches should work just as well.

Reply:
Ah the semicolon :-)